nomad/structs/secure_variables.go

```diff
@@ -69,92 +69,92 @@ type SecureVariableData struct {
 	KeyID string // ID of root key used to encrypt this entry
 }
 
-func (sv SecureVariableData) Copy() *SecureVariableData {
+// SecureVariableDecrypted structs are returned from the Encrypter's decrypt
+// method. Since they contains sensitive material, they should never be
+// persisted to disk.
+type SecureVariableDecrypted struct {
+	SecureVariableMetadata
+	SecureVariableItems
```

Comment:
One thing I'm not wild about with this struct is embedding the items rather than having their own name makes the resulting JSON API weird. I pulled this branch down locally to demonstrate what I mean:

```
$ echo '{"Items": {"foo": "bar"}}' | nomad operator api -X PUT /v1/var/jobs/httpd/web/http
Secure variable missing required Items object.%

$ echo '{"SecureVariableItems": {"foo": "bar"}}' | nomad operator api -X PUT /v1/var/jobs/httpd/web/http

$ nomad operator api /v1/var/jobs/httpd/web/http | jq .
{
  "CreateIndex": 12,
  "CreateTime": "2022-06-14T13:29:28.309635999Z",
  "ModifyIndex": 12,
  "ModifyTime": "2022-06-14T13:29:28.309635999Z",
  "Namespace": "default",
  "Path": "jobs/httpd/web/http",
  "SecureVariableItems": {
    "foo": "bar"
  }
}
```

I think we want this to be something like:

```go
type SecureVariableDecrypted struct {
	SecureVariableMetadata
	Items *SecureVariableItems
}
```